components/NavigationDrawer/examples/BottomDrawerExample.swift

```diff
@@ -0,0 +1,353 @@
+class BottomDrawerWithHeaderExample: UIViewController {
+
+  let headerViewController = DrawerHeaderViewController()
+  let contentViewController = DrawerContentViewController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showWrappedViewController()
+  }
+
+  func showWrappedViewController() {
+    let bottomDrawerViewController = MDCBottomDrawerViewController()
+    bottomDrawerViewController.mainContentViewController = contentViewController
+    bottomDrawerViewController.headerViewController = headerViewController
+    present(bottomDrawerViewController, animated: true, completion: nil)
+  }
+
+}
+
+class BottomDrawerNoHeaderExample: UIViewController {
+
+  let contentViewController = DrawerContentViewController()
+  let bottomDrawerTransitionController = MDCBottomDrawerTransitionController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showContentViewControllerOnly()
+  }
+
+  func showContentViewControllerOnly() {
+    contentViewController.transitioningDelegate = bottomDrawerTransitionController
+    contentViewController.modalPresentationStyle = .custom
+    present(contentViewController, animated: true, completion: nil)
+  }
+
+}
+
+class DrawerContentTableViewController: UITableViewController {
+
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width: view.bounds.width, height: tableView.contentSize.height)
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.purple
+    self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+    self.tableView.isScrollEnabled = false
+  }
+
+  override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
+    cell.textLabel?.text = "cell #\(indexPath.item)"
+    cell.backgroundColor = .white
+    print(cell.textLabel?.text ?? "")
+    return cell
+  }
+
+  override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return 100
+  }
+
+  override func numberOfSections(in tableView: UITableView) -> Int {
+    return 1
+  }
+}
+
+class DrawerContentWithScrollViewController: UIViewController,
+    UICollectionViewDelegate, UICollectionViewDataSource {
+
+  let collectionView: UICollectionView
+  let layout = UICollectionViewFlowLayout()
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width: view.bounds.width,
+                    height: collectionView.collectionViewLayout.collectionViewContentSize.height)
+    }
+    set {
```

Comment:
Can this setter be omitted if it just calls the super?  Does Swift require the pass-through override?

Reply:
it cannot afaik, unless it is a read-only property.